### PR TITLE
feat: Adding rules for subscriptions in avm/res/service-bus/namespace

### LINKED ISSUE #2657: [AVM Question/Feedback]: servicebus/namespace/topics/subcriptions/rules not availble?
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

I'm rewriting some of my bicep code to use AVM as much as possible. For my service bus namespace I'm setting a few rules via the default module 'Microsoft.ServiceBus/namespaces/topics/subscriptions/rules@2022-10-01-preview'
I can't find this in the service-bus/namespace module in AVM. Is there a plan to add this?

### PR DESCRIPTION
**Pull request by timdhaeyer**
## Description

Adding rules for subscriptions in avm/res/service-bus/namespace

## Pipeline Reference

| Pipeline |
| -------- |
|[avm.res.service-bus.namespace](https://github.com/timdhaeyer/bicep-registry-modules/actions/workflows/avm.res.service-bus.namespace.yml)          |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by timdhaeyer** [comment]
@microsoft-github-policy-service agree company="Zure"

**Comment by ChrisSidebotham** [comment]
Hey @timdhaeyer - This looks promising, I believe you also used the helper script. can you update the readme by running the set-avmmodule toolset as described here: https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/

Then run the workflow to test the new configuration in your subscription?

**Comment by timdhaeyer** [comment]
@ChrisSidebotham I have done what you requested. I added a test file that just creates a single rule inside a subscription. You can find it under the test files. Is this ok for you? :-)

**Comment by ChrisSidebotham** [comment]
@timdhaeyer - that looks alot better. Are you able to enable and run the workflow in your fork? you can then attach the status to the pipeline badges at the top of this PR?

**Comment by timdhaeyer** [comment]
@ChrisSidebotham Maybe a stupid question. But what workflow should I run? :-)

**Comment by timdhaeyer** [comment]
[avm.res.service-bus.namespace](https://github.com/timdhaeyer/bicep-registry-modules/actions/workflows/avm.res.service-bus.namespace.yml)

**Comment by ChrisSidebotham** [comment]
Hey @timdhaeyer - sorry for the delay here, we made some changes last week. Can you just sync your local version and rerun the workflow. the `Set-AVMModule` may need rerunning.

File: avm/res/service-bus/namespace/README.md
Comment (AlexanderSehr) [inline avm/res/service-bus/namespace/README.md:1]: Please note, the `main.json` templates **must** be re-build with the latest Bicep version `0.31.34` as this version (running on the agents) renders a different `main.json` file then previous versions. 

The steps to execute would be
- Upgrade your Bicep CLI version to `0.31.34` (e.g. via `choco upgrade bicep` if choco is used)
- Re-run the `Set-AVMModule` script on your module. E.g. via `Set-AVMModule -ModuleFolderPath '.\avm\res\automation\automation-account\' -SkipFileAndFolderSetup -SkipReadMe -Recurse`
- Push the changes
- Re-run the tests to ensure everything works. 

If the static tests continue to fail for you, please note that there is currently presumably another new bug which affects some of our modules ([ref](https://github.com/Azure/bicep/issues/15517)).

**Comment by jtracey93** [comment]
Hey @timdhaeyer ,

Firstly, thanks for your work on this PR!

We have made some changes to the AVM CI, detailed below, which means we need you to update your fork to pull in these latest changes and re-run your tests to show they still are passing prior to approving and merging this PR, as we don't and it fails once merged the publishing of your module will fail and will be blocked going forward until the test pass again via additional PRs.

### Changes to CI That Have Been Made That You Need To Take Action On

- https://github.com/Azure/bicep-registry-modules/pull/3745
    - Adds WAF Security PSRule Checks using this custom baseline of rules ([`/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml))
    - **Action Required By You:**
        - Ingest/pull-in these changes into your PR branch and re-run your tests to ensure they pass
            - If any issues with the rules and you think it should apply, please ping/reach out to the core team

Any questions reach out to the AVM Core Team by tagging us in your PR here or internally via Teams

Thanks

Jack (AVM Core Team)

**Comment by AlexanderSehr** [comment]
Closing in favor of #4106.
Note @timdhaeyer, the other PR is merely a rebase of this one with some smaller changes. You'll continue to be listed as the author of the changes you proposed in this PR.

**Comment by timdhaeyer** [comment]
Thanks @AlexanderSehr! Sorry to have kept this open for so long. The end of the year was a bit too busy for me.

**Comment by AlexanderSehr** [comment]
> Thanks @AlexanderSehr! Sorry to have kept this open for so long. The end of the year was a bit too busy for me.

That, plus most folks are off. So no worries 😄

**Pull request closed without merging**